client/components/data/query-memberships/index.js

```diff
@@ -15,39 +15,24 @@ import { requestProducts } from 'state/memberships/product-list/actions';
 
 class QueryMemberships extends Component {
 	static propTypes = {
-		productId: PropTypes.number,
 		siteId: PropTypes.number.isRequired,
-		requestProduct: PropTypes.func,
 		requestProducts: PropTypes.func,
 	};
 
 	static defaultProps = {
 		productId: null,
 	};
 
-	UNSAFE_componentWillMount() {
-		this.request( this.props );
-	}
-
-	UNSAFE_componentWillReceiveProps( nextProps ) {
-		if ( nextProps.siteId !== this.props.siteId || nextProps.productId !== this.props.productId ) {
-			this.request( nextProps );
-		}
-	}
-
-	request( props ) {
-		const { siteId, productId } = props;
-
-		if ( ! siteId ) {
+	componentDidMount() {
+		if ( this.props.requesting ) {
 			return;
 		}
 
-		// Products are indexed from 1.
-		if ( productId === 0 ) {
+		if ( ! this.props.siteId ) {
 			return;
 		}
 
-		props.requestProducts( siteId );
+		this.props.requestProducts( this.props.siteId );
```

Comment:
if this triggers an API call to get products for a given site then I don't think we need or want to eliminate the times when the `siteId` changes. on that note it's encouraged to use `componentDidUpdate()`

that is, it probably makes more sense for you to leave the `request( { siteId } )` function in tact but call it from `componentDidMount()` and `componentDidUpdate()`

we typically want to avoid code which triggers re-renders from inside `componentDidUpdate()` but API calls are different, especially if they maintain their own freshness requirements. that is, if we `setState()` then we can trigger a re-render loop that can freeze the app, but since API calls operate on such long scales and shouldn't be built based upon the information they provide (vs. based on the props passed from the parent) then they should be singular occurrences.

Reply:
Should be good now :)